### PR TITLE
ztp: added new tests for policyGenerator

### PR DESCRIPTION
**Pull request by nishant-parekh**
ztp: added new tests for policyGenerator
    - Added new tests covering the policyBuilder logic
    - Added tests for day0 and day2 artifacts for ZTP
    - Coverage of over 85% on the policyBuilder logic
    Signed-off-by: Nishant Parekh <nparekh@redhat.com>

### CODE REVIEW

File: ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGenerator_test.go [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+		parentDir = path.Join(getOutPath(t), dir)
+		dirName = file
+	}
+	assert.True(t,checkDirectoryExists(parentDir, dirName))
+	var dirPath = path.Join(parentDir, dirName)
+	assert.True(t, checkFileExists(dirPath, fileName))
+	placementRuleAndBindingAssertions(dirPath, t)
+	var ACMResourceFilePath = path.Join(dirPath,fileName)
+	checkIfSourceYamlEqualsACMObjectDefinition(ACMResourceFilePath, t)
+
```
Comment (irinamihai) [inline ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGenerator_test.go:132]: This line is extra.

Comment (lack) [reply ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGenerator_test.go:132]: We should always run our go code through `gofmt` to fixup the formatting.  In fact, this is now enforced on all GitHub PRs :)

File: ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/testFiles/TestPtpConfig/sourcePolicies/TestPtpConfig.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
+apiVersion: ptp.openshift.io/v1
```
Comment (serngawy) [inline ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/testFiles/TestPtpConfig/sourcePolicies/TestPtpConfig.yaml:1]: Is there a reason to duplicate the source files ? can we just refer to what is already under ztp/source-policy-crs ?

**Review by serngawy — CHANGES_REQUESTED** [review]
Please remove the changes in .gitignore file it is not owned by ZTP owners. we really need to merge this PR it takes long time.

File: ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/testFiles/TestPtpConfig/sourcePolicies/TestPtpConfig.yaml (hunk 2 above)
Comment (nishant-parekh) [reply ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/testFiles/TestPtpConfig/sourcePolicies/TestPtpConfig.yaml:1]: trying to keep the CRs we use for testing separate.. as we would eventually add invalid CRs as well to go over failure cases.. try and maintain distinction between configuration and test config..

File: ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGenerator_test.go (hunk 1 above)
Comment (nishant-parekh) [reply ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGenerator_test.go:134]: updated after running `gofmt`

**Pull request merged**